subsys/bluetooth/mesh/transport.c

```diff
@@ -1506,3 +1506,52 @@ void bt_mesh_rpl_clear(void)
 	BT_DBG("");
 	(void)memset(bt_mesh.rpl, 0, sizeof(bt_mesh.rpl));
 }
+
+void bt_mesh_heartbeat_send(void)
+{
+	struct bt_mesh_cfg_srv *cfg = bt_mesh_cfg_get();
+	u16_t feat = 0U;
+	struct __packed {
+		u8_t  init_ttl;
+		u16_t feat;
+	} hb;
+	struct bt_mesh_msg_ctx ctx = {
+		.net_idx = cfg->hb_pub.net_idx,
+		.app_idx = BT_MESH_KEY_UNUSED,
+		.addr = cfg->hb_pub.dst,
+		.send_ttl = cfg->hb_pub.ttl,
+	};
+	struct bt_mesh_net_tx tx = {
+		.sub = bt_mesh_subnet_get(cfg->hb_pub.net_idx),
+		.ctx = &ctx,
+		.src = bt_mesh_model_elem(cfg->model)->addr,
+		.xmit = bt_mesh_net_transmit_get(),
+	};
+
+	hb.init_ttl = cfg->hb_pub.ttl;
+
+	if (bt_mesh_relay_get() == BT_MESH_RELAY_ENABLED) {
```

Comment:
Is that correct? Shouldn't the current cfg->hb_pub.feat status be returned? Does not mean that any test case is failing because of this but I'm just asking.

As written in 4.3.2.63 Config Hearbeat Publication Status:
"The Features field shall identify the Heartbeat Publication Features state (see Section 4.2.17.5)."

Reply:
@MariuszSkamra I always understood cfg->hb_pub.feat as the triggers for sending the message, but not for controlling the content of the message. I'll need to double-check the spec and test spec.

Reply:
4.2.17.5 Heartbeat Publication Features: "The Heartbeat Publication Features state determines the features that trigger sending Heartbeat messages when changed."

4.3.2.63 Config Heartbeat Publication Status: "Bit field indicating features that trigger Heartbeat messages when changed"

So from that perspective this code looks correct.